src/main/java/com/ioet/bpm/people/boundaries/PersonController.java

```diff
@@ -51,17 +61,29 @@ public PersonController(PersonRepository personRepository) {
 
     @PutMapping("/{id}")
     public ResponseEntity<Person> updatePerson(@PathVariable(value = "id") String personId,
-                                             @Valid @RequestBody Person personToUpdate) {
+                                               @Valid @RequestBody Person personToUpdate) throws UnsupportedEncodingException, NoSuchAlgorithmException {
 
         Optional<Person> personFound = personRepository.findById(personId);
-
         if (personFound.isPresent()) {
-            personToUpdate.setId(personId);
+            personToUpdate.setId(personFound.get().getId());
+            personToUpdate.setPassword(personFound.get().getPassword());
             Person updatedPerson = personRepository.save(personToUpdate);
             return new ResponseEntity<>(updatedPerson, HttpStatus.OK);
         }
         return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
 
+    @PutMapping("/change-password/{id}")
```

Comment:
This should be a PostMapping, I agree you're updating something but the main idea is to do an action:
`Change Password`.